[No Ticket][Risk=no] Add messaging around unique error code

blrubenstein opened the pull request:
Description:
This adds copy to clipboard functionality for the unique error codes for triage.

Looks like this. Verbiage is not final, because this is still sorta unstyled. 

<img width="1920" alt="Screen Shot 2019-12-18 at 3 03 22 PM" src="https://user-images.githubusercontent.com/15351021/71119297-d9780d00-21a7-11ea-9db5-7327d029b99e.png">



---
**PR checklist**

- [ ] This PR meets the Acceptance Criteria in the JIRA story
- [ ] The JIRA story has been moved to Dev Review
- [ ] This PR includes appropriate unit tests
- [ ] I have run and tested this change locally
- [ ] If this includes an API change, I have updated the appropriate Swagger definitions and notified API consumers
- [ ] If this includes a new feature flag, I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later

jaycarlton commented on ui/src/app/components/error-handler.tsx:
```diff
@@ -98,12 +112,20 @@ export const ErrorHandler = withGlobalError()(class extends React.Component<Prop
+          <FlexColumn>
+          Server Error (500)
+          <div>
+            Please click to copy unique code for support ticket
```
Also, I bet there's a Zendesk Create Ticket request/URL that can take parameters like this. If so, this could be moved into an Open Ticket feature.

jaycarlton commented on ui/src/app/components/error-handler.tsx:
```diff
@@ -98,12 +112,20 @@ export const ErrorHandler = withGlobalError()(class extends React.Component<Prop
+          Server Error (500)
+          <div>
+            Please click to copy unique code for support ticket
+            {globalError.errorUniqueId && <CopyToClipboard text={globalError.errorUniqueId}>
```
Yeah, I think you'll need to make this conditioned on the `errorUniqueId` being non-null and non-empty.

blrubenstein replied:
Sure! Will do.

blrubenstein commented on ui/src/app/utils/errors.tsx:
```diff
@@ -37,9 +37,10 @@ export async function convertAPIError(e) {
       message = null,
       statusCode = null} = await e.json();
-    return { errorClassName, errorCode, message, statusCode };
+    return { errorClassName, errorCode, errorUniqueId, message, statusCode };
```
This will default to null if it isn't present. Which should be fine. (That's the above `errorUniqueId = null`)

jaycarlton replied:
👍

blrubenstein commented:
@jaycarlton I think I addressed all your feedback. I also added a little 'copied to clipboard' tooltip on click, so the user knows it did something. PTAL when you are back :D

jaycarlton commented:
Looks good. Thanks for doing this.

jaycarlton commented:
Looks good. Thanks for doing this.
